### PR TITLE
#257 allow to reuse variables from other yaml files

### LINKED ISSUE #257: Allow to reuse variables from other YAML files
Currently the functional logic (loops, variables, ...) only work on a per YAML-file basis. It is not possible to declare variables once in one YAML file and use it then in multiple others. This is often desirable though for more complex setups.

Currently the code in https://github.com/Netcentric/accesscontroltool/blob/65f4dfd962a525a0ac1ea2e5f4433f07c5a5eed6/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/configreader/YamlMacroProcessorImpl.java#L59 leads to the fact that for each YAML file the variables map is being empty.

In addition one would need to clarify the order in which YAML files are processed (depth first vs. breadth first or all in alphabetical order).

### PR DESCRIPTION
**Pull request by ghenzler**
Closes #257

### CODE REVIEW

**Comment by ghenzler** [comment]
includes #490 and should be merged after

File: accesscontroltool-bundle/src/test/java/biz/netcentric/cq/tools/actool/configreader/YamlMacroProcessorTest.java [hunk 1]
```diff
@@ -481,5 +477,75 @@ public void testDefWithUnderscore() throws Exception {
...
+        for(int i=0; i<expectedVals.length && it.hasNext(); i++) {
+            AceBean aceBean = it.next();
+            assertEquals("Expected path for bean "+i, expectedVals[i][0], aceBean.getJcrPath());
+            assertEquals("Expected initial content for bean "+i, expectedVals[i][1], aceBean.getInitialContent());
+        }
+    
+
+    }
+    
 }
```
Comment (kwin) [inline accesscontroltool-bundle/src/test/java/biz/netcentric/cq/tools/actool/configreader/YamlMacroProcessorTest.java:581]: Can we have a test for global variables as well?

Comment (ghenzler) [reply accesscontroltool-bundle/src/test/java/biz/netcentric/cq/tools/actool/configreader/YamlMacroProcessorTest.java:581]: Added one in a1feaba

File: docs/AdvancedFeatures.md [hunk 2]
```diff
@@ -215,19 +218,21 @@ There is also a multi-line variant of the DEF statement that allows to define co
       - name: "Value ${simpleMap[loopVar]}"
 ```
 
+### Global variables 
+
 By default, variables are local, this means the scope of a variable is limited to:
  * the lines in the very same yaml file following the definition till it is either redefined or the end of the yaml file is reached 
  * `FOR` loop in which variable is defined or re-defined.
```
Comment (kwin) [inline docs/AdvancedFeatures.md:225]: Please add a sentence about precedence of variables (I guess same name local variable takes precedence) and also add tests for same name local and global variables

Comment (ghenzler) [reply docs/AdvancedFeatures.md:225]: See 1fec561

**Review by kwin — APPROVED** [review]

**Pull request merged**